File: ABC/299/B_Trick_Taking.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
int main(){
    int N, T, num = 0, ans = 0; cin >> N >> T;
    vector<int> C(N), R(N);
    for(int i = 0; i < N; i++) cin >> C[i];
    for(int i = 0; i < N; i++) cin >> R[i];
    bool flag = false;
    for (int i = 0; i < N; i++) {
        if (C[i] == T) {
            flag = true;
        }
    }
    if (flag) {
        for (int i = 0; i < N; i++) {
            if (C[i] == T && R[i] > num) {
                ans = i+1;
                num = R[i];
            }
        }
    }
    else {
        for (int i = 0; i < N; i++) {
            if (C[i] == C[0] && R[i] > num) {
                ans = i+1;
                num = R[i];
            }
        }
    }
    cout << ans << endl;
    return 0;
}

File: ABC/299/B_Trick_Taking.py
N, T = map(int, input().split())
C = list(map(int, input().split()))
R = list(map(int, input().split()))

flag = False
value = 0
ans = 0
for i in range(N):
    if C[i] == T:
        flag = True

if flag:
    for i in range(N):
        if C[i] == T and value < R[i]:
            ans = i+1
            value = R[i]

else:
    for i in range(N):
        if C[i] == C[0] and value < R[i]:
            ans = i+1
            value = R[i]


print(ans)